patches/minecraft/net/minecraft/world/effect/MobEffect.java.patch

```diff
@@ -17,6 +17,16 @@
     }
  
     public Optional<MobEffectInstance.FactorData> m_216881_() {
+@@ -56,7 +_,8 @@
+          }
+       } else if (this == MobEffects.f_19614_) {
+          if (p_19467_.m_21223_() > 1.0F) {
+-            p_19467_.m_6469_(p_19467_.m_269291_().m_269425_(), 1.0F);
++            // Neo: Replace DamageSources#magic() with forge:poison to allow differentiating poison damage.
++            p_19467_.m_6469_(p_19467_.m_269291_().m_269079_(net.minecraftforge.common.ForgeMod.POISON_DAMAGE), 1.0F);
```

Comment:
How is this supposed to deal with Modded to Vanilla communication? Does poison damage default to magic if it is not registered on the associated side?

Reply:
Vanilla communication works fine as the damage types are synced - internally, this new type has the same definition as magic damage and will behave exactly the same.

I modified the code to allow poison to kill a player, connected with a vanilla client, and can observe no changes from a vanilla perspective:
![](https://i.imgur.com/XORoLTF.png)

Reply:
What happens if a datapack is installed in the server that changes the tag contents? Does that then not create a disconnect between what the vanilla client thinks needs to happen and what the server calculates?

Reply:
No, the damage calculations are done entirely serverside. Honestly I'm surprised DamageType is synced at all.